### PR TITLE
[7X] Skip ssh if segments are on the same host with coordinator.

### PR DESCRIPTION
**Pull request by higuoxing**
We can skip ssh connections if segments are on the same host with the coordinator. We can save some time when executing commands. E.g., with this patch, we don't ssh connection when creating the demo cluster and the creation time can be reduced from 18s to 10s on my laptop. Besides, we can skip setting up ssh keys for deploying the demo cluster.

## Here are some reminders before you submit the pull request
- [x] Pass `make installcheck`

### CODE REVIEW

**Comment by beeender** [comment]
With the patch, our local dev process will be a bit different from the production env for utilities. Possibly, some connection-related bugs could not be exposed earlier. But we do have cluster tests to cover the connection part.

I am uncertain about this. Let's see what others think.

File: gpMgmt/bin/gppylib/commands/base.py [hunk 1]
```diff
@@ -504,14 +505,19 @@ def execute(self, cmd, pickled=False, start_new_session=False):
...
+                         "{targethost} \"{gphome} {cmdstr}\"".format(targethost=self.targetHost,
+                                                                     gphome=". %s/greenplum_path.sh;" % self.gphome,
+                                                                     cmdstr=cmd.cmdStr)
+        else:
+            cmd.cmdStr = "bash -c \"{gphome} {cmdstr}\"".format(gphome=". %s/greenplum_path.sh;" % self.gphome,
+                                                                cmdstr=cmd.cmdStr)
         LocalExecutionContext.execute(self, cmd, pickled=pickled, start_new_session=start_new_session)
         if (cmd.get_stderr().startswith('ssh_exchange_identification: Connection closed by remote host')):
             self.__retry(cmd, 0, pickled)
-        pass
```
Comment (yihong0618) [inline gpMgmt/bin/gppylib/commands/base.py:514]: nice

File: gpMgmt/bin/gppylib/commands/base.py [hunk 2]
```diff
@@ -25,6 +25,7 @@
 import subprocess
 import sys
 import time
+import socket
```
Comment (yihong0618) [inline gpMgmt/bin/gppylib/commands/base.py:28]: import socket before import time to make it sort maybe better

**Review by ashwinstar — COMMENTED** [review]
> With the patch, our local dev process will be a bit different from the production env for utilities. Possibly, some connection-related bugs could not be exposed earlier. But we do have cluster tests to cover the connection part.

makes quite a lot of difference, saving dev time is critical.
I am sure this will help cut down some time in ICW runs as well.
```
$ time ssh -o StrictHostKeyChecking=no -o ServerAliveInterval=60 127.0.0.1 echo "abcd"
abcd

real	0m0.202s
user	0m0.009s
sys	0m0.003s

$ time echo "abcd"
abcd

real	0m0.000s
user	0m0.000s
sys	0m0.000s
```

To differently mirror @beeender concern dev box and ICW runs are fine but the way code change is being made - it will get applied to single container based CLI tests as well in CI which is definitely not desirable. We do run same tests in multi-vm setup still don't feel right testing CLI code with this change, I am fine ICW tests and cluster creation utilizing this speedup.

**Review by yihong0618 — APPROVED** [review]
LGTM

**Pull request merged**